app/models/subject.rb

```diff
@@ -2,21 +2,24 @@
 #
 # Table name: subject
 #
-#  created_at   :datetime
-#  id           :bigint           not null, primary key
-#  subject_code :text
-#  subject_name :text
-#  type         :text
-#  updated_at   :datetime
+#  created_at      :datetime
+#  id              :bigint           not null, primary key
+#  subject_area_id :bigint
+#  subject_code    :text
+#  subject_name    :text
+#  type            :text
+#  updated_at      :datetime
 #
 # Indexes
 #
-#  index_subject_on_subject_name  (subject_name)
+#  index_subject_on_subject_area_id  (subject_area_id)
+#  index_subject_on_subject_name     (subject_name)
 #
 
 class Subject < ApplicationRecord
   has_many :course_subjects
   has_many :courses, through: :course_subjects
+  belongs_to :subject_area, foreign_key: :type, inverse_of: :subjects
```

Comment:
I don't see why this needs an `inverse_of`, and as demonstrated below, why introduce an additional place where something can be done wrong.

```suggestion
  belongs_to :subject_area, foreign_key: :type
```